src/halodrops/processor.py

```diff
@@ -967,9 +986,32 @@ def add_q_and_theta_to_l2_ds(self):
         else:
             ds = self.l2_ds
 
-        ds = calc_q_from_rh(ds)
-        ds = calc_theta_from_T(ds)
+        ds = hh.calc_q_from_rh(ds)
+        ds = hh.calc_theta_from_T(ds)
 
         object.__setattr__(self, "_interim_l3_ds", ds)
 
         return self
+
+
+@dataclass(order=True, frozen=True)
+class Gridded:
+    sondes: dict
+    # platforms: dict = field(default_factory=dict)
+    _: KW_ONLY
+    interpolation_grid: np.ndarray = field(
+        default_factory=lambda: np.arange(0, 10010, 10)
+    )
+    interpolation_bins: np.ndarray = field(
+        default_factory=lambda: np.arange(-5, 10015, 10).astype("int")
+    )
+    max_gap_fill: int = 50
+
+    def concat_sondes(self):
+        """
+        function to concatenate all sondes using the combination of all measurement times and launch times
+        """
+        list_of_l2_ds = [sonde.l2_ds for sonde in self.sondes.values()]
```

Comment:
```suggestion
        list_of_l2_ds = [sonde._interim_l3_ds for sonde in self.sondes.values()]
```

The attribute `l2_ds` should stay unchanged because that will be the access to L2 data. Instead anything modified on that can be in the attr `_interim_l3_ds`, which will be mutable and the eventual state of that attribute should be what `Gridded` wants to concatenate.